### PR TITLE
Installation instructions

### PR DESCRIPTION
**Pull request by jcarceol**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,74 @@
+## Installation Instructions
+
+Follow these steps to clone the ArduPilot repository, set up your environment, and configure it for the CubeOrangeXL-DID autopilot.
```
Comment (amilcarlucas) [inline README.md:3]:
```suggestion
Follow these steps to clone the ArduPilot repository, set up your environment, and configure it for a specific flight controller hardware board.
```

File: README.md [hunk 2]
```diff
@@ -1,3 +1,74 @@
...
+   sudo apt update
+   sudo apt install git
+   ```
+
+2. **Python 3** (used by `waf` build system):
+   ```bash
+   sudo apt install python3 python3-pip
+   ```
+
+3. **ARM GCC Toolchain** (required to compile ArduPilot for ARM-based boards like CubeOrange):
```
Comment (amilcarlucas) [inline README.md:20]:
```suggestion
3. **ARM GCC Toolchain** (required to compile ArduPilot for ARM-based boards):
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,74 @@
...
+   ```bash
+   sudo ln -s /usr/bin/python3 /usr/bin/python
+   ```
+
+### Clone the Repository
+
+1. **Clone the repository with the required branch**:
+   Clone the `pt3.1-dcp` branch, including all submodules:
+   ```bash
+   git clone --branch pt3.1-dcp --recurse-submodules https://github.com/Reblade/ardupilot.git
```
Comment (amilcarlucas) [inline README.md:36]:
```suggestion
   Usually you either want a stable branch like `Copter-4.5` or `Plane-4.5` or you want the latest development branch `master`.
   Clone the desired branch, including all submodules, `Copter-4.5` in this example:
   ```bash
   git clone --branch Copter-4.5 --recurse-submodules https://github.com/ArduPilot/ardupilot.git
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,74 @@
...
+   Navigate into the `ardupilot` directory:
+   ```bash
+   cd ardupilot
+   ```
+
+### Configure the Build
+
+1. **Configure the build system for the CubeOrangeXL-DID board**:
+   ```bash
+   ./waf configure --board CubeOrangeXL-DID
```
Comment (amilcarlucas) [inline README.md:49]:
```suggestion
1. **Configure the build system for the desired flight controller board, in this example the `CubeOrangeXL-DID` board**:
   ```bash
   ./waf configure --board CubeOrangeXL-DID
```

File: README.md [hunk 5]
```diff
@@ -1,3 +1,74 @@
...
+   ```
+
+   This step will set up the environment and ensure that the build system is properly configured for the target autopilot board.
+
+### Build and Upload Firmware (Optional)
+
+Once configured, you can build and upload the firmware to the CubeOrangeXL autopilot:
```
Comment (amilcarlucas) [inline README.md:56]:
```suggestion
Once configured, you can build and upload the firmware to the flight controller:
```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,74 @@
...
+   ```bash
+   ./waf copter
+   ```
+
+2. **Upload the firmware to the autopilot** (if connected via USB):
+   ```bash
+   ./waf copter --upload
+   ```
+
+Now you have successfully cloned, configured, and built the ArduPilot firmware for the CubeOrangeXL-DID autopilot.
```
Comment (amilcarlucas) [inline README.md:68]:
```suggestion
Now you have successfully cloned, configured, and built the ArduPilot firmware for your flight controller.
```

File: README.md [hunk 7]
```diff
@@ -1,3 +1,74 @@
+## Installation Instructions
+
+Follow these steps to clone the ArduPilot repository, set up your environment, and configure it for the CubeOrangeXL-DID autopilot.
+
+### Prerequisites
```
Comment (amilcarlucas) [inline README.md:5]:
```suggestion
### Prerequisites for Linux
```

**Review by amilcarlucas — CHANGES_REQUESTED** [review]
Please change it, it can not go in like this

**Comment by peterbarker** [comment]
Please also note https://ardupilot.org/dev/docs/git-interactive-rebase.html .  If that's too hard for you I can do that bit once the content's right

**Review by khancyr — CHANGES_REQUESTED** [review]
No, that won't do. Please just add a link to the wiki, those instructions are wrong and opinionated. thankx

File: README.md [hunk 8]
```diff
@@ -1,3 +1,74 @@
...
+   ```
+
+2. **Python 3** (used by `waf` build system):
+   ```bash
+   sudo apt install python3 python3-pip
+   ```
+
+3. **ARM GCC Toolchain** (required to compile ArduPilot for ARM-based boards like CubeOrange):
+   ```bash
+   sudo apt install gcc-arm-none-eabi gdb-arm-none-eabi
```
Comment (khancyr) [inline README.md:22]: that isn't our toolchain. don't use it

**Pull request closed without merging**